### ISSUE TITLE
Change the interpretation of `_variable`

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#5036]
Right now `_variable` just seems to be interpreted as `_` except that they can't be used twice. But i think this is suboptimal. Take this case
```
pub def f(): Int32 = 42
pub def _f(): Int32 = 42
```
I don't think that this should be allowed, but it is.

Now look at this example:
```
pub def _f(): Int32 = 42
pub def g(): Int32 = f()
```
I get an error saying that `f` is undefined. I think I should get the error that `_f` cannot be called with a `_` prefix.

Another example. These `_variable` things are difficult with recursion.
```
pub def _f(): Int32 = _f()
pub def g(): Int32 = g()
```
The top line is actually okay, but I don't think this `_variable` thing should require renaming each instance. It is more natural IMO to see `_f` as the symbol `f` that cannot be used. I think that this should be allowed (by allowing the function itself to circumvent this rule)
```
pub def _f(): Int32 = f()
```
And note that the `f` in the body cannot reference some other thing because of the first point of the thread.

**Comment by magnus-madsen** [comment]
I am partial to the first part. At least we should discuss it.

For `let_ x` is big problems. We have many binders for variables, including channel select and Datalog constraints. I don't think we should mess with keywords, the `_` belongs on the variable. This also makes sense since it is the variable that is unused.

**Comment by JonathanStarup** [comment]
> I am partial to the first part. At least we should discuss it.
> 
> For `let_ x` I see big problems. We have many binders for variables, including channel select and Datalog constraints. I don't think we should mess with keywords, the `_` belongs on the variable. This also makes sense since it is the variable that is unused.

I agree that `let_ x = ...` seems too weird, I just wanted to point out that maybe, if `_f` can be referenced by `f`, the underscore shouldn't be attached to the variable. But I am mostly interested in the first part
